pkg/unified/e2e/test_helpers.go

```diff
@@ -0,0 +1,147 @@
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	machinev1 "github.com/openshift/api/machine/v1beta1"
+	"github.com/openshift/cluster-api-actuator-pkg/pkg/unified"
+	"github.com/openshift/cluster-api-actuator-pkg/pkg/unified/backends"
+	awsv1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TestHelper provides common helper functions for unified framework tests.
+type TestHelper struct {
+	uf          *unified.UnifiedFramework
+	client      runtimeclient.Client
+	ctx         context.Context
+	platform    configv1.PlatformType
+	machineSpec interface{}
+}
+
+// NewTestHelper creates a new test helper instance.
+func NewTestHelper(ctx context.Context, uf *unified.UnifiedFramework, client runtimeclient.Client, platform configv1.PlatformType, machineSpec interface{}) *TestHelper {
+	return &TestHelper{
+		uf:          uf,
+		client:      client,
+		ctx:         ctx,
+		platform:    platform,
+		machineSpec: machineSpec,
+	}
+}
+
+// CreateTemplate creates and validates a machine template.
+func (th *TestHelper) CreateTemplate(name string) interface{} {
+	tpl, err := th.uf.CreateMachineTemplate(th.ctx, th.client, th.platform, backends.BackendMachineTemplateParams{
+		Name:     name,
+		Platform: th.platform,
+		Spec:     th.machineSpec,
+	})
+	Expect(err).NotTo(HaveOccurred(), "Should create Machine Template")
+
+	return tpl
+}
+
+// CreateMachineSet creates and validates a machine set.
+func (th *TestHelper) CreateMachineSet(name string, template interface{}, labels map[string]string) interface{} {
+	ms, err := th.uf.CreateMachineSet(th.ctx, th.client, backends.BackendMachineSetParams{
```

Comment:
nit: Can we have the variable names a bit longer for readability.
ms->machineSet
uf -> framework
tpl -> template above